components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1124,7 +1124,7 @@ public void createApplicationTemplateFromSP(ServiceProvider serviceProvider, SpT
                     for (String msg : e.getValidationMsg()) {
                         log.error(msg);
                     }
-                    throw e;
+                    throw new IdentityApplicationManagementException(StringUtils.join(e.getValidationMsg(), ' '), e);
```

Comment:
e.getValidationMsg() returns an array of all the validation error messages. So that, joined all those error messages to a String, separated with spaces.